app/src/main/java/com/example/localieapp/ConsumerDealsFragment.kt

```diff
@@ -43,47 +46,65 @@ class ConsumerDealsFragment : Fragment() {
         arguments?.let {
             param1 = it.getString(ARG_PARAM1)
             param2 = it.getString(ARG_PARAM2)
-            coupons = ArrayList<Coupon>()
+//            coupons = it.getParcelableArrayList<Coupon>("coupons")
         }
     }
 
```

Comment:
Add some comments here about how the coupons are being populated. We will need to change this in the future to take into account user preferences